sbin/spark-daemon.sh

```diff
@@ -78,6 +78,12 @@ spark_rotate_log ()
     if [ -n "$2" ]; then
 	num=$2
     fi
+
+    expr $num + 0 &>/dev/null
+    if [ $? -ne 0 ] || [ $num -eq 0 ]; then
```

Comment:
Are you trying to verify if `num` is a number? I get it but you would already get a `bash` error in `$num -eq 0` if so and we don't otherwise validate script input that way. Isn't this forcing num to be nonzero, when I thought the issue was handling the case of 0 correctly?